src/OddJobs/Job.hs

```diff
@@ -455,7 +462,18 @@ jobMonitor = do
 
 -- | Ref: 'jobPoller'
 jobPollingSql :: Query
-jobPollingSql = "update ? set status = ?, locked_at = ?, locked_by = ?, attempts=attempts+1 WHERE id in (select id from ? where (run_at<=? AND ((status in ?) OR (status = ? and locked_at<?))) ORDER BY attempts ASC, run_at ASC LIMIT 1 FOR UPDATE) RETURNING id"
+jobPollingSql =
+  "update ? set status = ?, locked_at = ?, locked_by = ?, attempts=attempts+1\
+  \ WHERE id in (select id from ? where (run_at<=? AND ((status in ?) OR (status = ? and locked_at<?))) \
+  \ ORDER BY attempts ASC, run_at ASC LIMIT 1 FOR UPDATE) RETURNING id"
+
+jobPollingWithResourceSql :: Query
+jobPollingWithResourceSql =
+  " UPDATE ? SET status = ?, locked_at = ?, locked_by = ?, attempts = attempts + 1 \
+  \ WHERE id in (select id from ? where (run_at<=? AND ((status in ?) OR (status = ? and locked_at<?))) \
+  \ AND ?(id) \
+  \ ORDER BY attempts ASC, run_at ASC LIMIT 1) \
+  \ RETURNING id"
```

Comment:
is this the same as jobPollingSql?

Reply:
No - the key difference is the line `AND ?(id)`, which is where the are-resources-free function is interpolated. If we wanted to unify the two queries, we could create a dummy function in the DB that always returns `true` (and mark it `IMMUTABLE`, to make Postgres optimise it), and interpolate that function here in the case where resources aren't being used for concurrency control.

Reply:
I think it's fine to keep them separate, they just looked really similar.